rust/parser/src/lexer.rs

```diff
@@ -831,19 +832,45 @@ impl<'alloc> Lexer<'alloc> {
             }
         }
         builder.push_matching('/');
+        let mut flag_text = AutoCow::new(&self);
         while let Some(ch) = self.peek() {
             match ch {
                 '$' | '_' | 'a'..='z' | 'A'..='Z' | '0'..='9' => {
                     self.chars.next();
                     builder.push_matching(ch);
+                    flag_text.push_matching(ch);
                 }
                 _ => break,
             }
         }
 
+        // 12.2.8.2.1 Assert literal is a RegularExpressionLiteral.
+        let literal = builder.finish(&self);
+
+        // 12.2.8.2.2 Check that only gimsuy flags are mentioned at most once.
+        let gimsuy_mask : u32 = ['g', 'i', 'm', 's', 'u', 'y'].into_iter()
+            .map(|x| 1 << ((*x as u8) - ('a' as u8))).sum();
```

Comment:
According to https://godbolt.org/z/r_SYrn this compiles to a constant. There is something a little outrageous about it, or maybe I'm just old-fashioned.